File: #5 - Clean Architecture Version (Cubit)/lib/features/todo/domain/entities/todo_entity.dart
import 'package:flutter/foundation.dart' show immutable;

@immutable
class TodoEntity {
  const TodoEntity({
    this.id,
    required this.userId,
    required this.title,
    required this.dueOn,
    required this.status,
  });

  final int? id;
  final int userId;
  final String title;
  final DateTime dueOn;
  final TodoStatus status;
}

enum TodoStatus {
  completed(true),
  pending(false),
  all(false);

  final bool value;

  const TodoStatus(this.value);
}

File: gorest-backend/src/main/kotlin/com/sinasys/gorest/backend/entity/TodoEntity.kt
package com.sinasys.gorest.backend.entity


import jakarta.persistence.Id
import java.time.LocalDateTime
import jakarta.persistence.Table
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import com.sinasys.gorest.backend.dto.TodoStatus


@Entity
@Table(name = "todo")
open class TodoEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long = 0,

    @Column(nullable = false)
    open var userId: Long = 0,

    @Column(nullable = false)
    open var title: String = "",

    @Column(name = "due_on")
    open var dueOn: LocalDateTime? = null,

    @Enumerated(EnumType.STRING)
    open var status: TodoStatus = TodoStatus.Pending
)